common/src/main/java/com/wynntils/models/characterstats/CharacterStatsModel.java

```diff
@@ -158,4 +160,9 @@ public List<GearInfo> getWornGear() {
     private void centerSegmentCleared() {
         powderSpecialSegment.replaced();
     }
+
+    @SubscribeEvent
+    public void onHeldItemChanged(ChangeCarriedItemEvent event) {
+        powderSpecialSegment.replaced();
```

Comment:
How is this a special case? I assume this is related to item handler's rename canceling?

Reply:
Currently the powder segment resets the charge data to 0 if we get the coordinates or something like that, but it is possible to not get the coordinates when switching weapons. But switching weapons always resets to 0. So this accounts for it.

Reply:
Please add a short comment explaining this.

(In general, when someone asks a question in a review, it means the code was not clear to an (informed) reader. Hence, the reply should be stored with the source code, if future (informed) readers have the same question, not just in the PR.)